cmake/options.cmake

```diff
@@ -0,0 +1,11 @@
+#
+# Copyright (c) 2021-present, Trail of Bits, Inc.
+# All rights reserved.
+#
+# This source code is licensed in accordance with the terms specified in
+# the LICENSE file found in the root directory of this source tree.
+#
+
+option(ANVILL_ENABLE_INSTALL_TARGET "Set to ON to enable the install directives")
+option(ANVILL_ENABLE_TESTS "Set to ON to enable the tests")
+option(ANVILL_INSTALL_PYTHON3_LIBS "Install Python 3 libraries. Requires ANVILL_ENABLE_INSTALL_TARGET" ON)
```

Comment:
Should we add the `USE_SYSTEM_DEPENDENCIES` option here (found in `cmake/vcpkg_helper.cmake`)?

We don't want to officially support using your own dependencies, but for anyone who wants to try packaging themselves, the option should be a bit more discoverable (although we do print a message when we can't find vcpkg)
```
set(USE_SYSTEM_DEPENDENCIES OFF CACHE BOOL "Use system dependencies instead of trying to find vcpkg")
``

Reply:
With standard CMake practices, not passing anything should attempt to use system dependencies; I think we should build/test this specific configuration on the CI if we want to document it as a viable build procedure

Reply:
When I added `USE_SYSTEM_DEPENDENCIES`, it was under the assumption that the only supported build method would be to use vcpkg, and if anyone forgot to pass `VCPKG_ROOT`, there would be an ERROR telling them to add it. I knew this was not standard practice, but there had been so many issues with building in the past that I thought forcing the correct and supported way would be easiest. 

However, in order to at least give the option to distro packagers (who are probably a bit more advanced with CMake knowledge and able to debug things themselves) to test building with their own dependencies, I added the `USE_SYSTEM_DEPENDENCIES` option so they don't have to figure out what to do or patch the lifting tools themselves.

I don't think we need to test the `USE_SYSTEM_DEPENDENCIES` in CI (at least not extensively on all LLVM versions) if we aren't going to officially support it, and the logic around should be a simple conditional check that excludes all built-in vcpkg handling.